wai-middleware-prometheus/src/Network/Wai/Middleware/Prometheus.hs

```diff
@@ -64,12 +66,35 @@ instrumentApp handler app req respond = do
     start <- getCurrentTime
     app req $ \res -> do
         end <- getCurrentTime
-        let latency = fromRational $ toRational (end `diffUTCTime` start) * 1000000
-        let method = BS.unpack (Wai.requestMethod req)
-        let status = show (HTTP.statusCode (Wai.responseStatus res))
-        Prom.withLabel (handler, method, status) (Prom.observe latency) requestLatency
+        let method = Just $ BS.unpack (Wai.requestMethod req)
+        let status = Just $ show (HTTP.statusCode (Wai.responseStatus res))
+        observeMicroSeconds handler method status start end
         respond res
 
+-- | Instrument an IO action with timing metrics. This function can be used if
+-- you would like to get more fine grained metrics, for instance this can be
+-- used to instrument individual end points.
+--
+-- If you use this function you will likely want to override the default value
+-- of 'prometheusInstrumentApp' to be false so that your app does not get double
+-- instrumented.
+instrumentIO :: String  -- ^ The label used to identify this IO operation
+             -> IO a    -- ^ The IO action to instrument
+             -> IO a    -- ^ The instrumented app
+instrumentIO label io = do
+    start  <- getCurrentTime
+    result <- io
+    end    <- getCurrentTime
+    observeMicroSeconds label Nothing Nothing start end
+    pure result
```

Comment:
It looks like this is failing on GHC 7.8

src/Network/Wai/Middleware/Prometheus.hs:89:5: Not in scope: ‘pure’